http4s/src/main/scala/puretracing/http4s/Http4sTracer.scala

```diff
@@ -0,0 +1,26 @@
+package puretracing.http4s
+
+import cats.data.Kleisli
+import cats.effect.Sync
+import cats.syntax.all._
+import org.http4s.syntax.StringSyntax
+import org.http4s.{Header, HttpApp}
+import puretracing.api.Propagation
+import puretracing.cats.dsl._
+
+object Http4sTracer extends StringSyntax {
+
+  private[http4s] val spanHeader = "trace-id"
+
+  def apply[F[_]: Propagation: Sync](http: HttpApp[F]): HttpApp[F] =
```

Comment:
Is it possible to define this as `def apply[F[_]: Sync](http: HttpApp[F])(implicit T: Tracer[F]): HttpApp[ReaderT[T.Span]]` ?

As this is the place that we should generate a Span out of thin air (or from headers) i.e call `Tracer.startRootSpan`

Reply:
Don't take my work for the type siganture, I am probably completely wrong. I only wanted to point out that we should call `createRootSpan` rather than `currentSpan`.